providers/flagd/pkg/service/service.go

```diff
@@ -193,66 +380,173 @@ func handleError(err error) openfeature.ResolutionError {
 	return openfeature.NewGeneralResolutionError(err.Error())
 }
 
-// EventStream emits received events on the given channel
-func (s *Service) EventStream(
-	ctx context.Context, eventChan chan<- *schemaV1.EventStreamResponse, maxAttempts int, errChan chan<- error,
-) {
-	delay := s.baseRetryDelay
-	var err error
-	for i := 1; i <= maxAttempts; i++ {
-		s.logger.V(logger.Debug).Info("attempt at connection to event stream", "attempt", i)
-		i, err = s.eventStream(ctx, eventChan, i)
-		if i == 1 {
-			delay = s.baseRetryDelay // reset delay if the connection was successful before failing
-		}
-		if err != nil && i <= maxAttempts {
+func (s *Service) EventChannel() <-chan of.Event {
+	return s.events
+}
+
+// startEventStream - starts listening to flagd event stream with retries.
+// If retrying is exhausted, an event with openfeature.ProviderError is emitted.
+func (s *Service) startEventStream(ctx context.Context) {
```

Comment:
maybe worth adding a line saying that this function is expected to be called in a go routine

Reply:
Good suggestion. Done - https://github.com/open-feature/go-sdk-contrib/pull/317/commits/3eee6b23362073325def650e01a5973d831bae74